### PR TITLE
[0.9] Remove `env` from Trace APIs and delete FlowState* APIs

### PR DESCRIPTION
**Pull request by ssbushi**

### CODE REVIEW

File: go/genkit/servers.go
Comment (ssbushi) [inline go/genkit/servers.go:1]: I'm not very confident on the changes in golang. This passes tests but I can skip these changes too if that is preferred.

One caveat is that the reflection API is already changed in this PR so it makes little sense to have golang use the old API.

File: genkit-tools/common/src/api/reflection.ts [hunk 1]
```diff
@@ -78,8 +78,8 @@ registry.registerPath({
 });
 registry.registerPath({
```
Comment (pavelgj) [inline genkit-tools/common/src/api/reflection.ts:79]: can just remove this, it's gone

File: genkit-tools/common/src/api/reflection.ts [hunk 2]
```diff
@@ -96,8 +96,8 @@ registry.registerPath({
...
```
Comment (pavelgj) [inline genkit-tools/common/src/api/reflection.ts:97]: can just remove this, it's gone

File: go/genkit/servers.go
Comment (pavelgj) [reply go/genkit/servers.go:1]: leave go as is, @apascal07 is working on the telemetry server changes

File: samples/js-coffee-shop/src/index.ts [hunk 3]
```diff
@@ -28,9 +28,7 @@ configureGenkit({
     dotprompt(),
   ],
   enableTracingAndMetrics: true,
-  flowStateStore: 'firebase',
```
Comment (pavelgj) [inline samples/js-coffee-shop/src/index.ts:31]: samples are still on 0.5... can leave them as is for now

**Review by pavelgj — APPROVED** [review]

File: go/genkit/servers.go
Comment (apascal07) [reply go/genkit/servers.go:1]: Can leave or revert, I can rebase, it's not a big deal either way.

Comment (ssbushi) [reply go/genkit/servers.go:1]: I'll revert just to be safe. Thank you for clarifying.

**Comment by tonybaroneee** [comment]
@ssbushi this appears to have broken trace viewing. Not sure what's going on and why the server is still expecting an env param, trying to get to the bottom of it...

image

**Comment by ssbushi** [comment]
@tonybaroneee Can I get a repro? I tried this locally and it seems to work (`next` for the reflection API and `redesign` for the Dev UI)

**Comment by tonybaroneee** [comment]
@ssbushi All good now!

**Pull request merged**